src/averageAges.js

```diff
@@ -15,11 +15,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person =>
+    century
+      ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+      : person.sex === 'm');
+
+  const ages = men.map(person => person.died - person.born);
+
+  return ages.reduce((x, y) => x + y) / ages.length;
```

Comment:
```suggestion
  return men.reduce((sum, person) =>(
    sum + (person.died - person.born)
  ),0) / ages.length;
```

Reply:
also move it in the separate function and you this function here and in the next function